packages/listview/src/components/AuditLogs/filterDropdown/filters.tsx

```diff
@@ -1,126 +1,102 @@
 import React, { useContext, useRef } from 'react';
 import { Option, ButtonIcon } from 'react-rainbow-components';
 import { TrashFilled } from '@rainbow-modules/icons';
 import { StyledFilterContainer, StyledLabel, StyledPicklist, StyledInput } from './styled';
 import { LabelFilter as LabelFilterType, PicklistValue } from '../types';
 import context from '../context';
-import getLabelFilters from './helpers/getLabelFilters';
 
-const FilterOptions = ({ labels = [] }: { labels: string[] }) => (
-    <>
-        {labels.map((label) => (
-            <Option name={label} label={label} />
-        ))}
-    </>
-);
+const FilterOptions = ({ labels: activeLabels = [] }: { labels: string[] }) => {
+    const { labels } = useContext(context);
+    return (
+        <>
+            {labels
+                .filter((label) => !activeLabels.includes(label))
+                .map((label) => (
+                    <Option name={label} label={label} />
+                ))}
+        </>
+    );
+};
 
 const LabelFilter = ({
+    index,
     name: nameInProps,
     value: valueInProps,
-    index,
     label,
-}: LabelFilterType & { index: number }) => {
-    const { filters, updateFilters, labels } = useContext(context);
+    filteredLabels,
+    onChange,
+}: {
+    name: string;
+    value?: string;
+    index: number;
+    label: string;
+    filteredLabels: string[];
+    onChange: (oldName: string, newName?: string, value?: string) => void;
+}) => {
+    const { labels } = useContext(context);
     const inputRef = useRef<HTMLInputElement>();
 
-    const handleNameChange = ({ name }: PicklistValue) => {
-        if (!name || typeof name !== 'string') return;
-        const { labels: filterLabels = {} } = filters;
-        const newLabels = { ...filterLabels };
-        if (Object.keys(filterLabels).includes(name)) {
-            newLabels[name].push(valueInProps);
-        } else {
-            newLabels[name] = [valueInProps];
-        }
-        newLabels[nameInProps] = newLabels[nameInProps].filter((val) => val !== valueInProps);
-        if (newLabels[nameInProps].length === 0) {
-            delete newLabels[nameInProps];
-        }
-        const newFilters = {
-            ...filters,
-            labels: newLabels,
-        };
-        updateFilters(newFilters);
-        setTimeout(() => inputRef.current?.focus());
-    };
+    const handleNameChange = ({ name }: PicklistValue) =>
+        onChange(nameInProps, name as string, valueInProps);
 
     const handleValueChange = (event: React.ChangeEvent<HTMLInputElement>) => {
         const newValue = event.target.value;
-        const newLabels: Record<string, string[]> = { ...filters.labels };
-        const newValues = [...newLabels[nameInProps]];
-        newValues[index] = newValue;
-        newLabels[nameInProps] = newValues;
-        updateFilters({
-            ...filters,
-            labels: newLabels,
-        });
+        onChange(nameInProps, nameInProps, newValue);
     };
```

Comment:
I see below you define this:
```onChange: (oldName: string, newName?: string, value?: string) => void;```
I think we should use an object thus we have better understanding of the prop names